field/src/extension/mod.rs

```diff
@@ -10,3 +13,47 @@ pub trait BinomiallyExtendable<const D: usize>: Field + Sized {
     const W: Self;
     fn ext_multiplicative_group_generator() -> [Self; D];
 }
+
+/// Optimally extendable field trait.
+/// An bionomial extension field with a prime base field.
+pub trait OptimallyExtendable<const D: usize>: BinomiallyExtendable<D> + PrimeField {
+    // DTH_ROOT = W^((p - 1)/D)
+    const DTH_ROOT: Self;
+}
+
+pub trait Frobenius<F: OptimallyExtendable<D>, const D: usize>:
```

Comment:
I think we should support Frobenius in some more general way, like either a `frobenius` method in `Field` itself, or maybe a `Frobenius: Field` trait to make it optional

Reply:
I see your point. The frobenuis map for a general field needs to specify a characteristic p, which looks meaningless for a standard field(typically, for the prime field, it just returns itself).  For extension fields like complex or recently defined extensions, we could define the p as the order of the base field. Our current algorithm requires us to compute the `Dth-root` for the constant in the binomial poly, so we only compute a linear multiplication. If we make it more general, it can be difficult to implement the default algorithm. 

For now, I can think about one way to generalize it; instead of using a prime field, we can extend it to a general field, but the base field still needs some trait like a ForbenuisExtendable to specify the root of `w^(n^D-1/n)`. Then, the extended field will not be an OEF but a BEF that supports frobenuis. Do you think this is OK?

Reply:
Extend the definition to a non-prime base field. Since the field is no longer an OEF, rename it back to BEF. The current implementation requires the base field to define the frobenuis.